sdk/src/main/scala/com/horizen/account/storage/AccountStateMetadataStorageView.scala

```diff
@@ -196,30 +196,33 @@ class AccountStateMetadataStorageView(storage: Storage) extends AccountStateMeta
   }
 
   private[horizen] def saveToStorage(version: ByteArrayWrapper): Unit = {
-    require(withdrawalEpochInfoOpt.nonEmpty, "WithdrawalEpochInfo must be NOT NULL.")
-    require(blockFeeInfoOpt.nonEmpty, "BlockFeeInfo must be NOT NULL.")
+
     require(accountStateRootOpt.nonEmpty, "Account State Root must be NOT NULL.")
 
     val updateList = new JArrayList[JPair[ByteArrayWrapper, ByteArrayWrapper]]()
     val removeList = new JArrayList[ByteArrayWrapper]()
 
-    // Update Withdrawal epoch related data
-    updateList.add(new JPair(withdrawalEpochInformationKey,
-      new ByteArrayWrapper(WithdrawalEpochInfoSerializer.toBytes(withdrawalEpochInfoOpt.get))))
+    withdrawalEpochInfoOpt.foreach(epochInfo => {
+      // Update Withdrawal epoch related data
+      updateList.add(new JPair(withdrawalEpochInformationKey,
+        new ByteArrayWrapper(WithdrawalEpochInfoSerializer.toBytes(epochInfo))))
+
+      // Update BlockFeeInfo data
+      val nextBlockFeeInfoCounter: Int = getBlockFeeInfoCounter(epochInfo.epoch) + 1
```

Comment:
I'm not sure that we need to put feeInfo processing inside `withdrawalEpochInfoOpt` block.
in case `withdrawalEpochInfoOpt` is not defined, we should take the existing one form the storage

Reply:
Ok, done